build.cjs

```diff
@@ -0,0 +1,647 @@
+ * @param {Boolean} [options.incremental=false] true if build output should be cached for repeated builds
+ * @param {Boolean} [options.write=true] true if build output should be written to disk. If false, the files that would have been written as in-memory buffers
+ * @returns
+ */
+async function buildCesiumJs(options) {
+  const css = globby.sync(cssFiles);
+
+  const buildConfig = esbuildBaseConfig();
+  buildConfig.entryPoints = ["Source/Cesium.js"];
+  buildConfig.bundle = true;
+  buildConfig.minify = options.minify;
+  buildConfig.sourcemap = options.sourcemap;
+  buildConfig.external = ["https", "http", "url", "zlib"];
+  buildConfig.plugins = options.removePragmas ? [stripPragmaPlugin] : undefined;
+  buildConfig.incremental = options.incremental;
+  buildConfig.write = options.write;
+  // print errors immediately, and collect warnings so we can filter out known ones
+  buildConfig.logLevel = "error";
+
+  // Build ESM
+  const result = await esbuild.build({
+    ...buildConfig,
+    format: "esm",
+    outfile: path.join(options.path, "index.js"),
+  });
+
+  handleBuildWarnings(result);
+
+  const results = [result];
+
+  // Copy and minify non-bundled CSS and JS
+  const cssAndThirdPartyConfig = esbuildBaseConfig();
+  cssAndThirdPartyConfig.entryPoints = [
+    "Source/ThirdParty/google-earth-dbroot-parser.js",
+    ...css, // Load and optionally minify css
+  ];
+  cssAndThirdPartyConfig.minify = options.minify;
+  cssAndThirdPartyConfig.loader = {
+    ".gif": "text",
+    ".png": "text",
+  };
+  cssAndThirdPartyConfig.sourcemap = options.sourcemap;
+  cssAndThirdPartyConfig.outdir = options.path;
+  await esbuild.build(cssAndThirdPartyConfig);
+
+  // Build IIFE
+  if (options.iife) {
+    const result = await esbuild.build({
+      ...buildConfig,
+      format: "iife",
+      globalName: "Cesium",
+      outfile: path.join(options.path, "Cesium.js"),
+    });
+
+    handleBuildWarnings(result);
+
+    results.push(result);
+  }
+
+  if (options.node) {
+    const result = await esbuild.build({
+      ...buildConfig,
+      format: "cjs",
+      platform: "node",
+      sourcemap: false,
+      outfile: path.join(options.path, "index.cjs"),
+    });
+
+    handleBuildWarnings(result);
+    results.push(result);
+  }
+
+  return results;
+}
+
+function filePathToModuleId(moduleId) {
+  return moduleId.substring(0, moduleId.lastIndexOf(".")).replace(/\\/g, "/");
+}
+
+const sourceFiles = [
+  "Source/**/*.js",
+  "!Source/*.js",
+  "!Source/Workers/**",
+  "!Source/WorkersES6/**",
+  "Source/WorkersES6/createTaskProcessorWorker.js",
+  "!Source/ThirdParty/Workers/**",
+  "!Source/ThirdParty/google-earth-dbroot-parser.js",
+  "!Source/ThirdParty/_*",
+];
+
+/**
+ * Creates a single entry point file, Cesium.js, which imports all individual modules exported from the Cesium API.
+ */
+function createCesiumJs() {
+  let contents = `export const VERSION = '${version}';\n`;
+  globby.sync(sourceFiles).forEach(function (file) {
```

Comment:
This file is split between using the synchronous versions of certain functions and asynchronous versions of others. I personally prefer synchronous ones, but I'm curious if you've thought about this disparity and whether if it's worth addressing (it may not be - just something I wanted to point out).

Reply:
I did see this. Many of the synchronous functions like `createCesiumJs` are from the original `gulpfile.cjs` and have remained unchanged, just ported to this new file. I've opted to change them as little as possible for the sake of avoiding side effects from this change.

Reply:
I did do some cleanup so that it is at least consistant within the new functions.